core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java

```diff
@@ -154,4 +156,48 @@ public static boolean isDSWDatasource( Domain domain ) {
     }
     return !isMetadataDatasource( domain );
   }
+
+  protected String isMetaDataSource( String id ) {
+    Domain domain;
+    try {
+      domain = metadataDomainRepository.getDomain( id );
+      if ( domain == null ) {
+        return null;
+      }
+    } catch ( Exception e ) { // If we can't load the domain then we MUST return false
+      return null;
+    }
+    return isMetadataDatasource( domain ) ? id : null;
+  }
+
+  protected int noOfThreads() {
```

Comment:
For clarify, normality, try not to use abbreviations for functions and class names. But "no" is the standard abbreviation for number, so it's okay.